### DISCUSSION TITLE
Eclipse IDE integration

### DISCUSSION THREAD

**Discussion by mauritius84** [beehive-lab/TornadoVM#307] [General]
Hello!
Could you please help me with Eclipse IDE integration. I have succesfully builded Tornado VM and made some tests from command lines. Now I'm trying to write some simple application in Eclipse IDE.

I have installed Eclipse 2023‑12 R and pointed to Tornado VM. What else I should to do?

In the documentation mentioned that I should run next commands:
`
$ mvn eclipse:eclipse
$ python scripts/eclipseSetup.py
`

But where I should execute those commands. What expected result of the execution?

**Comment by jjfumero** [comment]
Hi @mauritius84. I am not an Eclipse user any more, but I think the process should be similar to IntelliJ. So what we do is to take the java arguments printed when using this command:

```bash
$ tornado --printJavaFlags
... -server -XX:-UseCompressedOops -XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCI -XX:-UseCompressedClassPointers --enable-preview -Djava.library.path=/home/juan/manchester/tornado/tornado/bin/sdk/lib  --module-path .:/home/juan/manchester/tornado/tornado/bin/sdk/share/java/tornado -Dtornado.load.api.implementation=uk.ac.manchester.tornado.runtime.tasks.TornadoTaskGraph -Dtornado.load.runtime.implementation=uk.ac.manchester.tornado.runtime.TornadoCoreRuntime -Dtornado.load.tornado.implementation=uk.ac.manchester.tornado.runtime.common.Tornado -Dtornado.load.device.implementation.opencl=uk.ac.manchester.tornado.drivers.opencl.runtime.OCLDeviceFactory -Dtornado.load.device.implementation.ptx=uk.ac.manchester.tornado.drivers.ptx.runtime.PTXDeviceFactory -Dtornado.load.device.implementation.spirv=uk.ac.manchester.tornado.drivers.spirv.runtime.SPIRVDeviceFactory -Dtornado.load.annotation.implementation=uk.ac.manchester.tornado.annotation.ASMClassVisitor -Dtornado.load.annotation.parallel=uk.ac.manchester.tornado.api.annotations.Parallel  --upgrade-module-path /home/juan/manchester/tornado/tornado/bin/sdk/share/java/graalJars -XX:+UseParallelGC  @/home/juan/manchester/tornado/tornado/bin/sdk/etc/exportLists/common-exports @/home/juan/manchester/tornado/tornado/bin/sdk/etc/exportLists/opencl-exports --add-modules ALL-SYSTEM,tornado.runtime,tornado.annotation,tornado.drivers.common,tornado.drivers.opencl
```

This will enable all Java flags that TornadoVM needs + all paths to the JNI-shared libraries that connect with the accelerators. 
Note that the flags vary per backend installed. The one I copied shows my paths using the OpenCL backend.

You copy these flags in your project runner for the applications you want to launch.

**Reply by mauritius84** [reply]
Hello @jjfumero ! Thank you very much for your advice. This works well for Eclipse too.
